File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowedFilter = [False] * 26
        for c in allowed:
            allowedFilter[ord(c) - ord('a')] = True
        
        numAllowedWords = 0
        for i in range(len(words)):
            allowedWord = True
            for a in words[i]:
                if not allowedFilter[ord(a) - ord('a')]:
                    allowedWord = False
                    break
                
            if allowedWord:
                numAllowedWords += 1
            
        
        return numAllowedWords

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.c


int countConsistentStrings(char * allowed, char ** words, int wordsSize){
    
    bool allowedFilter[26] = {false};
    for (char *c = allowed; *c != '\0'; ++c) {
        allowedFilter[*c - 'a'] = true;
    }
    
    int numAllowedWords = 0;
    for (int i = 0; i < wordsSize; ++i) {
        bool allowedWord = true;
        for (char* a = words[i]; *a != '\0'; ++a) {
            if (!allowedFilter[*a - 'a']) {
                allowedWord = false;
                break;
            }
        }  
        if (allowedWord) {
            ++numAllowedWords;
        }
    }
    return numAllowedWords;
}